pkg/network/server.go

```diff
@@ -1139,7 +1178,13 @@ func (s *Server) handleHeadersCmd(p Peer, h *payload.Headers) error {
 	if s.blockFetcher.IsActive() {
 		return nil
 	}
-	return s.stateSync.AddHeaders(h.Hdrs...)
+	for _, header := range h.Hdrs {
+		err := s.stateSync.AddItems(header)
```

Comment:
You must use `AddItems(h.Hrs...)`, otherwise it's useless. This operation must be batched.